[Runtime Env]Add usage of expansion of certain PATH variables in env_vars (#34856)
ray-project/ray: doc/source/ray-core/handling-dependencies.rst

  - Example: ``"pytorch_p36"``

- ``env_vars`` (Dict[str, str]): Environment variables to set.  Environment variables already set on the cluster will still be visible to the Ray workers; so there is
  no need to include ``os.environ`` or similar in the ``env_vars`` field.   

>   By default, these environment variables will override the same name environment variables on the cluster. 

Reviewer: ```suggestion
  By default, these environment variables override the same name environment variables on the cluster. 
```

>   You can also reference existing environment variables using ${ENV_VAR} and achieve the append behavior.

Reviewer: ```suggestion
  You can also reference existing environment variables using ${ENV_VAR} to achieve the appending behavior.
```

>   If the user wants to append his value to the existing environment variable such as "PATH/LD_LIBRARY_PATH". he can refer to the example to set it.

Reviewer 2: You can also reference the existing environment variable via `${ENV_VAR}` and achieve the append behavior (currently only PATH,LD_LIBRARY_PATH,DYLD_LIBRARY_PATH,LD_PRELOAD are supported). See below for an example.

>   Currently, only PATH, LD_LIBRARY_PATH, DYLD_LIBRARY_PATH, and LD_PRELOAD are supported. See below for an example:

Reviewer: ```suggestion
  Only PATH, LD_LIBRARY_PATH, DYLD_LIBRARY_PATH, and LD_PRELOAD are supported. See below for an example:
```
